core/storage/btree.rs

```diff
@@ -35,6 +35,25 @@ const BTREE_HEADER_OFFSET_RIGHTMOST: usize = 8; /* if internalnode, pointer righ
 */
 pub const BTCURSOR_MAX_DEPTH: usize = 20;
 
+/// Evaluate a Result<CursorResult<T>>, if IO return IO.
+macro_rules! io {
+    ($expr:expr) => {
+        match $expr? {
+            CursorResult::Ok(v) => v,
+            CursorResult::IO => return Ok(CursorResult::IO),
+        }
+    };
+}
+
+/// Check if the page is unlocked, if not return IO.
+macro_rules! unlocked {
```

Comment:
I would suggest renaming it to something like return_if_locked to be more explicit to be honest.